apis/v1alpha2/shared_types.go

```diff
@@ -20,63 +20,96 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// GatewayAllowType specifies which Gateways should be allowed to use a Route.
-type GatewayAllowType string
-
-const (
-	// Any Gateway will be able to use this route.
-	GatewayAllowAll GatewayAllowType = "All"
-	// Only Gateways that have been  specified in GatewayRefs will be able to use this route.
-	GatewayAllowFromList GatewayAllowType = "FromList"
-	// Only Gateways within the same namespace as the route will be able to use this route.
-	GatewayAllowSameNamespace GatewayAllowType = "SameNamespace"
-)
+// ParentRef identifies an API object that should be considered a parent of this
+// resource. The only kind of parent resource with "Core" support is Gateway.
+// This API may be extended in the future to support additional kinds of parent
+// resources, such as HTTPRoute.
```

Comment:
Opinion: Until we come up with a story around how this extension point can be used, let's document it more conservatively to target a Gateway.

Reply:
Added some clarifications to state that the parent would usually be a Gateway here. I don't want to go so far as saying it would only be a Gateway.